subsys/net/ip/net_if.c

```diff
@@ -591,7 +591,7 @@ static void iface_router_expired(struct k_work *work)
 		}
 
 		iface_router_notify_deletion(router, "has expired");
-
+		sys_slist_find_and_remove(&active_router_timers, &router->node);
```

Comment:
since it is already looping on active_router_timers, you should be able to call sys_slist_remove() with the relevant changes if needed (previous node pointer needed ?)

Reply:
Good point, I'll rewrite it with sys_slist_remove()